### PR TITLE
[#4474] improvement(build): Support publish Gravitino through apache repo

### PR DESCRIPTION
**Pull request by yuqi1129**
### What changes were proposed in this pull request?

Replace maven publish repo sonatype with Apache repo.

### Why are the changes needed?

It's not proper to use sonatpe repo after Gravitino was denoted to ASF.

Fix: #4474 

### Does this PR introduce _any_ user-facing change?

N/A.

### How was this patch tested?

Locally.

image

### CODE REVIEW

**Comment by jerryshao** [comment]
@yuqi1129 please also check if the release stage repo can also be worked.

File: build.gradle.kts [hunk 1]
```diff
@@ -393,6 +372,25 @@ subprojects {
...
+          username = System.getenv("APACHE_USERNAME")
+          password = System.getenv("APACHE_PASSWORD")
+        }
+      }
+      maven {
+        name = "apacheReleases"
+        url = uri("https://repository.apache.org/service/local/staging/deploy/maven2/")
+        credentials {
+          username = System.getenv("APACHE_USERNAME")
+          password = System.getenv("APACHE_PASSWORD")
```
Comment (jerryshao) [inline build.gradle.kts:390]: How to choose to publish to snapshot repo or release repo?

**Comment by yuqi1129** [comment]
@jerryshao 
I have tried several times, and we could not find a solution to resolve the multiple-repositories problem temporarily. I reported this issue to the ASF, see: https://issues.apache.org/jira/browse/INFRA-26022

besides, I tried a different approach similar to how we publish to [Sonatype](https://s01.oss.sonatype.org/service/local/). 

```kotlin
nexusPublishing {
  repositories {
  	 create("apache") {
      val snapshot = uri("https://repository.apache.org/content/repositories/snapshots/")
      val release = uri("https://repository.apache.org/service/local/staging/deploy/maven2/")

      nexusUrl.set(release)
      snapshotRepositoryUrl.set(snapshot)
      username.set(System.getenv("APACHE_USERNAME"))
      password.set(System.getenv("APACHE_PASSWORD"))
      stagingProfileId.set("170bb01395e84a")
    }
    }
  }

  packageGroup.set("org.apache.gravitino")
}
```

However, publishing the snapshot jars to the apache repository is okay, it still have problem in publishing to the release  repository. The problem is as below:

```
Execution failed for task ':initializeApacheStagingRepository'.
> Failed to create staging repository, server at https://repository.apache.org/service/local/staging/deploy/maven2/ responded with status code 400, body: <html>
    <head>
      <title>400 - Bad Request</title>
      <meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
  
              
      <link rel="icon" type="image/png" href="https://repository.apache.org/favicon.png">
      <!--[if IE]>
      <link rel="SHORTCUT ICON" href="https://repository.apache.org/favicon.ico"/>
      <![endif]-->
  
      <link rel="stylesheet" href="https://repository.apache.org/static/css/Sonatype-content.css?2.14.20-02" type="text/css" media="screen" title="no title" charset="utf-8">
    </head>
    <body>
      <h1>400 - Bad Request</h1>
      <p>Cannot find a matching staging profile</p>
    </body>
  </html>
```

I am 100% certain that the staging profile ID is correct and there is no possible explanation for the error.

**Comment by jerryshao** [comment]
@yuqi1129 currently, we're using a different way (curl) to publish the maven jars, so we don't need to change this. Would you please close this PR, thanks.

**Comment by yuqi1129** [comment]
Closed it as another way to publish to maven repo has been proposed.

**Pull request closed without merging**